src/relay/transforms/fold_constant.cc

```diff
@@ -151,9 +151,12 @@ class ConstantFolder : public MixedModeMutator {
     }
 
     // We should think about potentially constant evaluation over these ops too.
-    if (call->op == invoke_tvm_op_ || call->op == shape_func_op_ || call->op == alloc_tensor_op_ ||
-        call->op == alloc_storage_op_ || call->op == device_copy_op_) {
-      return GetRef<Call>(call);
+    static auto fnoncomputational = Op::GetAttrMap<TNonComputational>("TNonComputational");
+    if (auto call_node = call->op.as<OpNode>()) {
```

Comment:
Let’s be more specific about the type although they are the same
```suggestion
    if (const auto* call_node = call->op.as<OpNode>()) {
```